### PR TITLE
Add support for `rgb-image` NeXus interpretation

### LINKED ISSUE #728: [NeXus] Add support for `rgb-image` interpretation
With the work on #649, we can now support the [rgb-image interpretation](https://manual.nexusformat.org/nxdl_desc.html#interpretation) of NeXus

### PR DESCRIPTION
**Pull request by axelboc**
Fix #728

image

### CODE REVIEW

File: src/h5web/providers/mock/metadata-utils.ts [hunk 1]
```diff
@@ -109,6 +109,13 @@ export function withAttributes<T extends Entity>(
   };
 }
 
+export function withImageAttributes<T extends Entity>(entity: T): T {
+  return withAttributes(entity, [
+    makeStrAttr('CLASS', 'IMAGE'),
+    makeStrAttr('IMAGE_VERSION', '1.2'),
+  ]);
+}
```
Comment (axelboc) [inline src/h5web/providers/mock/metadata-utils.ts:117]: Slightly simpler mock data API that supports NX datasets.

File: src/h5web/toolbar/RgbToolbar.tsx [hunk 2]
```diff
@@ -37,4 +37,4 @@ function RgbVisToolbar() {
   );
 }
 
-export default RgbVisToolbar;
+export default RgbToolbar;
```
Comment (axelboc) [inline src/h5web/toolbar/RgbToolbar.tsx:40]: Renaming this component for consistency.

File: src/h5web/toolbar/toolbars.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export { default as LineToolbar } from './LineToolbar';
+export { default as HeatmapToolbar } from './HeatmapToolbar';
+export { default as ComplexToolbar } from './ComplexToolbar';
+export { default as ComplexLineToolbar } from './ComplexLineToolbar';
+export { default as RgbToolbar } from './RgbToolbar';
```
Comment (axelboc) [inline src/h5web/toolbar/toolbars.ts:5]: Creating a couple of "index" files to clean up the imports in the vis-packs' vis definitions files.

File: src/h5web/vis-packs/core/visualizations.ts [hunk 4]
```diff
@@ -135,9 +139,9 @@ export const CORE_VIS: Record<Vis, CoreVisDef> = {
   [Vis.RGB]: {
     name: Vis.RGB,
     Icon: FiImage,
-    Toolbar: RgbVisToolbar,
+    Toolbar: RgbToolbar,
     Container: RgbVisContainer,
-    ConfigProvider: RgbVisConfigProvider,
+    ConfigProvider: RgbConfigProvider,
```
Comment (axelboc) [inline src/h5web/vis-packs/core/visualizations.ts:144]: Renaming this provider as well, again for consistency.

File: src/h5web/vis-packs/nexus/pack-utils.ts [hunk 5]
```diff
@@ -36,13 +42,20 @@ export function getSupportedVis(entity: Entity): VisDef | undefined {
...
+  const interpretation = getAttributeValue(dataset, 'interpretation');
+
+  if (
+    interpretation === NxInterpretation.RGB &&
+    hasNumDims(dataset, 3) &&
+    !isCplx
+  ) {
+    return NEXUS_VIS[NexusVis.NxRGB];
+  }
+
```
Comment (axelboc) [inline src/h5web/vis-packs/nexus/pack-utils.ts:55]: The code that matters the most in this PR.

File: src/h5web/vis-packs/nexus/visualizations.ts [hunk 6]
```diff
@@ -1,25 +1,33 @@
...
+  RgbConfigProvider,
+} from '../core/configs';
 
 export enum NexusVis {
   NxSpectrum = 'NX Spectrum',
+  NxComplexSpectrum = 'NX Complex Spectrum',
   NxImage = 'NX Image',
   NxComplexImage = 'NX Complex Image',
-  NxComplexSpectrum = 'NX Complex Spectrum',
+  NxRGB = 'NX RGB',
```
Comment (axelboc) [inline src/h5web/vis-packs/nexus/visualizations.ts:30]: I hesitated with `NxRgb`, like `RgbVis`. Let me know what you think.

File: src/h5web/vis-packs/nexus/pack-utils.test.ts [hunk 7]
```diff
@@ -103,6 +108,19 @@ describe('getSupportedVis', () => {
     expect(getSupportedVis(unknown)).toBe(NEXUS_VIS[NexusVis.NxImage]);
   });
 
+  it('should fall back when dataset is incompatible with interpretation', () => {
```
Comment (loichuder) [inline src/h5web/vis-packs/nexus/pack-utils.test.ts:111]: fall back to what ?

File: src/h5web/vis-packs/nexus/visualizations.ts (hunk 6 above)
Comment (loichuder) [reply src/h5web/vis-packs/nexus/visualizations.ts:30]: This is great :+1:

**Review by loichuder — APPROVED** [review]
All good :+1:

File: src/h5web/vis-packs/nexus/pack-utils.test.ts (hunk 7 above)
Comment (axelboc) [reply src/h5web/vis-packs/nexus/pack-utils.test.ts:111]: Same as the test above (the dataset shape), not sure it was worth repeating as it lengthens an already long test name ^^

Comment (loichuder) [reply src/h5web/vis-packs/nexus/pack-utils.test.ts:111]: Yeah but the sentence feels weird... I would prefer it to be repeated ^^

Comment (axelboc) [reply src/h5web/vis-packs/nexus/pack-utils.test.ts]: Fair enough 👍

**Pull request merged**